modernizer-annotation-processor/src/main/java/org/gaul/modernizer_annotation_processor/ModernizerAnnotationProcessor.java

```diff
@@ -0,0 +1,341 @@
+            className.append(parentClasses.get(index));
+            className.append("$");
+        }
+        return className.toString() + classElement.getSimpleName();
+    }
+
+    private String getMethodRepresentation(Element methodElement) {
+        ExecutableType method = getExecutableTypeElement(methodElement);
+        return ModernizerAnnotationUtils.getMethodRep(
+        getFullClassName(methodElement.getEnclosingElement()),
+        methodElement.getSimpleName().toString(),
+        getRepresentation(method.getReturnType()),
+        getParamsRepresentation(method.getParameterTypes(), methodElement));
+    }
+
+    private ExecutableType getExecutableTypeElement(Element methodElement) {
+        TypeVisitor<ExecutableType, Void> executableTypeVisitor =
+            new SimpleTypeVisitor6<ExecutableType, Void>() {
+                @Override
+                public ExecutableType visitExecutable(
+                    ExecutableType executableType, Void obj
+                ) {
+                    return executableType;
+                }
+            };
+        return methodElement.asType().accept(executableTypeVisitor, null);
+    }
+
+    /**
+     * Fetches the method parameters in the format the way ASM parses a method
+     * descriptor.
+     * @param methodParams A list of types of formal parameters
+     * of an executable type
+     * @param methodElement The element of executable type
+     * @return A list of all the parameters after formatting
+     * Example:
+     * Input: methodParams {@code {int[][], String, List<Integer>}}
+     * Output: {@code {"int[][]", "String", "java.util.List"}}
+     */
+    private List<String> getParamsRepresentation(
+        List<? extends TypeMirror> methodParams,
+        Element methodElement
+    ) {
+        List<String> methodParamReps = new ArrayList<String>();
+        /*
+         * For a non-static inner class constructor, adding the outer class
+         * object as an argument.
+         * Example:
+         * fullClassName = "org.gaul.example_package.OuterClass$InnerClass"
+         * methodParamReps.add("org.gaul.example_package.OuterClass")
+         */
+        if (methodElement.getKind().equals(ElementKind.CONSTRUCTOR)) {
+            String fullClassName =
+                getFullClassName(methodElement.getEnclosingElement());
+            Set<Modifier> modifiers =
+                methodElement.getEnclosingElement().getModifiers();
+            if (!modifiers.contains(Modifier.STATIC)) {
+                int index = fullClassName.lastIndexOf("$");
+                if (index != -1) {
+                    methodParamReps.add(fullClassName.substring(0, index));
+                }
+            }
+        }
+        for (TypeMirror param : methodParams) {
+            methodParamReps.add(getRepresentation(param));
+        }
+        return methodParamReps;
+    }
+
+    /**
+     * Fetching the parameter string by formatting type variables and
```

Comment:
I think the whole description could be `Returns an ASM-parsable string representation of the element.`